File: ios/rninternals/DeviceInfoModule.m
//
//  DeviceInfoModule.m
//  
//
//  Created by Hazo Baykulov on 07.07.2026.
//

#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_REMAP_MODULE(CustomDeviceInfo, DeviceInfoModule, NSObject)

RCT_EXTERN_METHOD(getModel)
RCT_EXTERN_METHOD(getSystemVersion)
RCT_EXTERN_METHOD(getBatteryLevel:(RCTPromiseResolveBlock)resolve
                  reject:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(getNativeStartTime)

@end

File: ios/rninternals/DeviceInfoModule.swift
import UIKit

@objc(DeviceInfoModule)
class DeviceInfoModule: NSObject {
  @objc
  func getModel() -> String {
      return UIDevice.current.model
  }

  @objc
  func getSystemVersion() -> String {
      return UIDevice.current.systemVersion
  }

  @objc
  func getBatteryLevel(_ resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
    UIDevice.current.isBatteryMonitoringEnabled = true
    let level = UIDevice.current.batteryLevel
    resolve(level * 100)
  }

  @objc
  func getNativeStartTime() -> NSNumber {
    return NSNumber(value: AppDelegate.nativeStartTimeMs)
  }
}